internal/extension/proto/extension.proto

```diff
@@ -0,0 +1,56 @@
+syntax = "proto3";
+
+package eg_extension;
+
+option go_package = "internal/extension/proto";
+
+// ObjectReference is the K8s API object referenced in an HTTPRouteFilter extensionRef.
+// The API object must be valid within the cluster and the Group and Kind must be
+// registered with the cluster.
+//
+// TODO: Gateway API currently doesn't allow setting a version...this is probably due to
+// the assumption that it is an internal implementation detail and should be resolved
+// be the implementation (complete guess or maybe we need to ask for the enhancement???)
+message ObjectReference {
+    // api_group is an K8s group of the referent (e.g gateway.networking.k8s.io).
+    string api_group = 1;
+
+    // kind is the Kind of the referent (e.g HTTPRoute, Service).
+    string kind = 2;
+
+    // name is the object name of the referent.
+    string name = 3;
+}
+
+// HTTPListenerExtensionContext provides context on
+// the users intent so that the provided XDS can be modified by the Extension.
+message HTTPListenerExtensionContext {
+    // route_filter_extension_ctxs provides a list of HTTPRouteFilters
+    // that use an extensionRef registered by the extension.
+    // The list is in the order it was processed in XDS Translation
+    repeated HTTPRouteFilterExtensionContext route_filter_extension_ctxs = 1;
+}
+
+// HTTPRouteFilterExtensionContext provides context about an HTTPRoute containing an
+// ExtensionRef HTTPRouteFilterType. The context provided includes which filterchain/s,
+// route and Group/Kind of the object referent.
+message HTTPRouteFilterExtensionContext {
+    // route_name is the name of HTTPRoute the filter is being applied on
+    //
+    // TODO: Will we need more context than just route name such as Path, Headers,
```

Comment:
On your comment: Lets explore trying to do this just with RouteName. Note: this does mean the Gateway project will need to ensure routes are named properly and unique which you will see that currently not all cases give a route a unique name

### Why? 
The reason for having EG as the foundation is so that everyone that wants to extend it doesn't have to duplicate work/algorithms. EG already handles the route matching so we just need to provide the Hook enough information to know which route the extensionRef is referring to.

If we find uses cases where that falls down or is sufficient then we can explore but for a v0 I want us to try and explore ways where we are not duplicating EG logic.

Reply:
yea think just matching by HTTPRoute name and hostname should be sufficient for now.

I guess it's going to depend how `HTTPRoutes are named in the IR. If it's by `namespace/HTTPRouteName` that should be enough to guarantee uniqueness because K8s doesn't allow you to have the same name for different objects of the same Kind within the same namespace. which looks like it does https://github.com/datawire/envoy-gateway/blob/8dd60b1a3aaa8a93504aae0efd4d50315a43bd96/internal/gatewayapi/helpers.go#L365.

This is appended in the ir.HTTPRoute.Name along with the host. https://github.com/datawire/envoy-gateway/blob/8dd60b1a3aaa8a93504aae0efd4d50315a43bd96/internal/gatewayapi/route.go#L260

Reply:
Yes, in the IR but it doesn't flow through to every Envoy Route that is created as seen here:

https://github.dev/envoyproxy/gateway/blob/c007313d7ec81d7728b847b62f1b513db93e4d73/internal/xds/translator/route.go#L38 

So, I think it works but we would have to modify it so the routes added to the VHost included the `.Name`. Now that probably gets tricky with mirroring and stuff now that I think about it....

Reply:
we've decided that `route_name` and `hostnames` are sufficient for the time being